src/core/tests/Test_CoreMain.py

```diff
@@ -432,6 +435,214 @@ def test_install_only_critical_and_security_packages_for_redhat_autopatching(sel
+        # check status file for configure patching assessment state
+        message = json.loads(substatus_file_data[0]["formattedMessage"]["message"])
+        self.assertEqual(message["autoAssessmentStatus"]["autoAssessmentState"], Constants.AutoAssessmentStates.ENABLED)  # auto assessment is enabled
+
+        # operation #2: Auto Assessment
+        argument_composer.activity_id = str(uuid.uuid4())
+        argument_composer.included_classifications_list = self.included_package_name_mask_list = self.excluded_package_name_mask_list = []
+        argument_composer.maintenance_run_id = None
+        argument_composer.start_time = runtime.env_layer.datetime.standard_datetime_to_utc(datetime.datetime.utcnow())
+        argument_composer.duration = Constants.AUTO_ASSESSMENT_MAXIMUM_DURATION
+        argument_composer.reboot_setting = Constants.REBOOT_NEVER
+        argument_composer.patch_mode = None
+        argument_composer.exec_auto_assess_only = True
+        runtime.execution_config.exec_auto_assess_only = True
+        CoreMain(argument_composer.get_composed_arguments())
+        # check telemetry events
+        self.__check_telemetry_events(runtime)
+        # check status file
+        with runtime.env_layer.file_system.open(runtime.execution_config.status_file_path, 'r') as file_handle:
+            status_file_data = json.load(file_handle)[0]["status"]
+        # verifying the original operation name is preserved
+        self.assertTrue(status_file_data["operation"] == Constants.CONFIGURE_PATCHING)
+        substatus_file_data = status_file_data["substatus"]
+        self.assertEquals(len(substatus_file_data), 2)
+        self.assertTrue(substatus_file_data[0]["name"] == Constants.PATCH_ASSESSMENT_SUMMARY)
+        self.assertTrue(substatus_file_data[0]["status"] == Constants.STATUS_SUCCESS.lower())
+        # check started by set to 'Platform'
+        self.assertTrue(json.loads(substatus_file_data[0]["formattedMessage"]["message"])['startedBy'], Constants.PatchAssessmentSummaryStartedBy.PLATFORM)
+        # verifying the older operation summary is preserved
+        self.assertTrue(substatus_file_data[1]["name"] == Constants.CONFIGURE_PATCHING_SUMMARY)
+        self.assertTrue(substatus_file_data[1]["status"] == Constants.STATUS_SUCCESS.lower())
+        message = json.loads(substatus_file_data[1]["formattedMessage"]["message"])
+        self.assertEqual(message["automaticOsPatchState"], Constants.AutomaticOsPatchStates.DISABLED)  # auto OS updates are disabled in RuntimeCompositor
+        # check status file for configure patching assessment state
+        message = json.loads(substatus_file_data[1]["formattedMessage"]["message"])
+        self.assertEqual(message["autoAssessmentStatus"]["autoAssessmentState"], Constants.AutoAssessmentStates.ENABLED)  # auto assessment is enabled
+
+        runtime.stop()
```

Comment:
Question: why is the `tearDown` method not used for this test class? I can see that most other test classes have it and use it

Reply:
We traditionally use teardown to discard any configuration, env setup needed for test execution. Whenever the env setup/config is common between multiple tests in a file we use the setup and teardown create and discard setup for each test.

The runtime configuration in this test file varies between different tests (see values set in argument_composer), so there each test is responsible for setting and discarding it's own environment. See comment on line 33 under setUp()

Reply:
Gotcha, thanks for the info! That makes sense.